### PR TITLE
Feature/api working spaces

### PR DESCRIPTION
**Pull request by Lex-23**
- created API for TypeWorkingSpace and WorkingSpace. View based on  mixins and Generic (just for compare with ApiView)
- wrote tests for API
- fixed validator
- added Swagger

### CODE REVIEW

File: coworkings_be/working_spaces/serializers.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from coworking.models import Coworking
+from coworking.serializers import CoworkingRelatedSerializer
+from rest_framework import serializers
+from working_spaces.models import TypeWorkingSpace, WorkingSpace
+
+
+class TypeWorkingSpaceSerializer(CoworkingRelatedSerializer):
```
Comment (andreyPromaster) [inline coworkings_be/working_spaces/serializers.py:7]: Just wondering, why have you decided to create CoworkingRelatedSerializer? I think it is not a good way to use this approach

File: coworkings_be/working_spaces/serializers.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    type = serializers.IntegerField()
+
+    class Meta:
+        model = TypeWorkingSpace
+        fields = "__all__"
+
+
+class WorkingSpaceSerializer(CoworkingRelatedSerializer):
+    coworking_queryset = Coworking.objects.all()
+    coworking = serializers.PrimaryKeyRelatedField(queryset=coworking_queryset)
```
Comment (andreyPromaster) [inline coworkings_be/working_spaces/serializers.py:19]:
```suggestion
    coworking = serializers.PrimaryKeyRelatedField(queryset=Coworking.objects.all())
```

File: coworkings_be/working_spaces/views.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+from utils.permissions import IsOwnerOrAdministratorRoleOrReadOnly
+from utils.validators import validate_request_to_coworking
+from working_spaces.models import TypeWorkingSpace, WorkingSpace
+from working_spaces.serializers import (
+    TypeWorkingSpaceSerializer,
+    WorkingSpaceSerializer,
+)
+
+
+class BaseWorkingSpaceListView(ListCreateAPIView):
```
Comment (andreyPromaster) [inline coworkings_be/working_spaces/views.py:16]: I think it is overly. I is better to split logic. In future a lot of validations points will be added. As result you will have to split logic in any case

File: coworkings_be/working_spaces/views.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            request.user, request.auth["user_role"], coworking.owner
+        )
+
+    def patch(self, request, *args, **kwargs):
+        self.validate_request(request)
+        return self.partial_update(request, *args, **kwargs)
+
+    def delete(self, request, *args, **kwargs):
+        self.validate_request(request)
+        return self.destroy(request, *args, **kwargs)
```
Comment (andreyPromaster) [inline coworkings_be/working_spaces/views.py:44]:
```suggestion
    def destroy(self, request, *args, **kwargs):
        self.validate_request(request)
        return super().destroy(request, *args, **kwargs)
```

File: coworkings_be/working_spaces/views.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        validate_request_to_coworking(
+            request.user, request.auth["user_role"], coworking.owner
+        )
+        return self.create(request, *args, **kwargs)
+
+
+class BaseWorkingSpaceDetailView(RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsOwnerOrAdministratorRoleOrReadOnly,)
+
+    def validate_request(self, request):
```
Comment (andreyPromaster) [inline coworkings_be/working_spaces/views.py:32]: It is not clear what this method is doing

Comment (Lex-23) [reply coworkings_be/working_spaces/views.py:32]: this method validate request

File: coworkings_be/working_spaces/views.py (hunk 4 above)
Comment (Lex-23) [reply coworkings_be/working_spaces/views.py:44]: but this approach not about drf generics.

File: coworkings_be/working_spaces/serializers.py (hunk 1 above)
Comment (Lex-23) [reply coworkings_be/working_spaces/serializers.py:7]: Why it is not a good way - to make general class for general logic?

**Review by andreyPromaster — APPROVED** [review]

**Comment by andreyPromaster** [comment]
Looks good, thanks

**Pull request merged**